src/Base/Writer.cpp

```diff
@@ -226,8 +226,16 @@ void Writer::addError(const std::string& msg)
     Errors.push_back(msg);
 }
 
-bool Writer::hasErrors() const
-{
+bool Writer::hasErrors()
+{
+    switch (errno) {
+        case ENOSPC:  // No space left
+        case EROFS:   // Read only
+        case ENODEV:  // No such device
+        case EACCES:  // Access denied
+            addError(strerror(errno));
+    }
+

```

Comment:
This feels like an awkward place to put this code: not only do you have to make a very `const`-sounding method into a non-`const`, but it doesn't address the use-case where someone might call `getErrors()` instead of `hasErrors()`.